core/src/mindustry/ai/BlockIndexer.java

```diff
@@ -78,7 +78,7 @@ public BlockIndexer(){
                 var drop = tile.drop();
 
                 if(drop != null){
-                    allOres.add(drop);
+                    allOres.put(drop, 0);
```

Comment:
What's the point of setting it to 0? The default value is 0 anyway.

Reply:
You could then check contains if needed

Reply:
> You could then check contains if needed for whatever reason

That's a pretty bad idea when you can just check `get(...) > 0`. I just don't see any reason for it to be here.

Reply:
Fair enough, will fix